ts/torch_handler/distributed/base_neuronx_continuous_batching_handler.py

```diff
@@ -0,0 +1,411 @@
+
+        if go_to_decode:
+            local_decode_seq_ids = torch.cat(torch.where(self.decode_seq_ids > -1))
+            local_decode_cache_ids = self.decode_cache_ids[local_decode_seq_ids]
+            local_decode_next_tokens = self.decode_next_tokens[local_decode_seq_ids]
+
+            local_next_tokens = self._run_decode(
+                local_decode_next_tokens, local_decode_cache_ids, local_decode_seq_ids
+            )
+
+            filter_prefill_seq_ids = (
+                torch.isin(local_decode_seq_ids, torch.as_tensor(prefill_seq_ids))
+                if len(prefill_seq_ids) > 0
+                else torch.full(local_decode_seq_ids.shape, False)
+            )
+
+            local_decode_cache_ids = local_decode_cache_ids + 1
+            for i, is_prefill_seq_id in enumerate(filter_prefill_seq_ids):
+                if not is_prefill_seq_id:
+                    seq_id = local_decode_seq_ids[i].item()
+
+                    if seq_id in req_decode_seq_ids:
+                        self._update_results(
+                            results,
+                            seq_id,
+                            i,
+                            local_decode_cache_ids,
+                            local_next_tokens,
+                        )
+                    else:
+                        req_id = self._get_req_id(seq_id)
+                        logger.warning(
+                            f"Found request id:{req_id} with seq_id:{seq_id} in local_decode_seq_ids, but not in batch requests. Delete it"
+                        )
+                        self._clean_up(seq_id, req_id)
+
+        return [results[i] for i in self.context.request_ids.values()]
+
+    def postprocess(self, inference_output):
+        self.context.stopping_criteria = [
+            self.context.cache[req_id]["stopping_criteria"]
+            for req_id in self.context.request_ids.values()
+        ]
+
+        return inference_output
+
+    def _get_empty_seq_id(self):
+        if len(self.empty_seq_ids) == 0:
+            # clean up dead req_ids due to client disconnction
+            self._clean_dead_reqs()
+
+        assert len(self.empty_seq_ids) > 0
+        return self.empty_seq_ids.pop()
+
+    def _add_empty_seq_id(self, seq_id):
+        self.empty_seq_ids.append(seq_id)
+
+    def _get_seq_id(self, req_id):
+        seq_id = None
+        cache = self.context.cache.get(req_id, None)
+        if cache:
+            seq_id = cache["seq_id"]
+        assert seq_id is not None, "{req_id} must have seq_id"
+        return seq_id
+
+    def _get_req_id(self, seq_id):
+        req_id = self.seq_id_to_req_id.get(seq_id, None)
+        assert req_id is not None
+        return req_id
+
+    def _pad_to_max(self, x):
+        z = torch.empty(x.shape[0], self.max_length, dtype=torch.int64)
+        for idx, item in enumerate(x):
+            pad = torch.zeros(self.max_length - len(x[idx]), dtype=torch.int)
+            z[idx] = torch.cat((x[idx], pad))
```

Comment:
we should test if stacking the concatenated tensors is faster

Reply:
the pad is added to the same dimension. Why stack is needed at here?